### PR TITLE
Feature/flake8

### PR DESCRIPTION
**Pull request by lillekemiker**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Community Node Submission

## Have you discussed this change with the InvokeAI team?
- [x] Yes
- [ ] No, because:

      
## Have you updated all relevant documentation?
- [ ] Yes
- [ ] No
- [x] N/A

## Description

As per my suggestions here https://github.com/invoke-ai/InvokeAI/discussions/3721
I am adding flake8 checks to GHA, to pre-commit hooks and also cleaning up flake8 issues.

## Related Tickets & Documents

- Related Issue #
- Closes #

## Added/updated tests?

- [ ] Yes
- [ ] No : _please replace this line with details on why tests
      have not been included_

### CODE REVIEW

**Review by lstein — APPROVED** [review]
This looks great. flake8 seems to have caught lots of problems. Better get this in before more conflicts accrue. I see a bunch already.

**Comment by lstein** [comment]
@lillekemiker I'm watching the conflicts build up. Are you planning to re-run `flake8` on current main and update the PR?

**Comment by lillekemiker** [comment]
> @lillekemiker I'm watching the conflicts build up. Are you planning to re-run `flake8` on current main and update the PR?

So flake8 only finds the errors, I have to correct them myself, unfortunately. But I don't mind.
@psychedelicious mentioned a couple of PRs that he wanted merged before this one, so I was mostly waiting for a go-ahead there. I forget which ones they were.
That being said, I can start addressing conflicts now so there are less when we are ready.

**Comment by psychedelicious** [comment]
@lillekemiker Ah, right - those have merged and the path is clear for this PR :)

**Comment by lillekemiker** [comment]
@psychedelicious @lstein - This should be ready merge as soon as the tests clear (which they already have locally)

File: scripts/orig_scripts/knn2img.py
Comment (psychedelicious) [inline scripts/orig_scripts/knn2img.py:394]: Does `flake8` catch reassignment like this? Strongly prefer non-mutating logic if possible; this makes me feel warm and fuzzy inside!

**Review by psychedelicious — APPROVED** [review]
Thanks! I'd like to get another set of eyes to glance over the changes before merging.

File: installer/lib/messages.py
Comment (psychedelicious) [inline installer/lib/messages.py:93]: I can't quite figure out what is going on here.

File: scripts/orig_scripts/knn2img.py
Comment (lillekemiker) [reply scripts/orig_scripts/knn2img.py:394]: Flake8 doesn't but mypy will. However, flake8 did catch this one for some reason that I don't exactly remember. I think it was complaining that a `Tensor` doesn't have an `.astype()` method maybe.

File: installer/lib/messages.py
Comment (lillekemiker) [reply installer/lib/messages.py:93]: Yeah, it's a bit less than obvious. It's two things. I'm removing an unnecessary extra call to `.expanduser().resolve()` since this was already done on `dest` and then I am also assigning the value to `init_path` which is used below but never assigned.

**Pull request merged**
